src/api/layers/ExternalWMSLayer.class.js

```diff
@@ -0,0 +1,59 @@
+import GeoAdminWMSLayer from '@/api/layers/GeoAdminWMSLayer.class'
+
+/**
+ * Metadata for an external WMS layer, we can use mostly what is already done in the class
+ * GeoAdminWMSLayer, except the ID definition. With an external WMS, we need to declare the ID as
+ * described in adr/2021_03_16_url_param_structure.md, not just the ID of the layer requested on the
+ * server.
+ */
+export default class ExternalWMSLayer extends GeoAdminWMSLayer {
```

Comment:
Why not use the `ExternalLayer` as base class?

Reply:
It was te re-use the `getURL()` generation, but I've now moved URL param management into the OpenLayers component (otherwise some param where doubled in the process). That frees me from having to use the GeoAdminWMSLayer class, it now inherits ExternalLayer